src/clusterfuzz/_internal/datastore/data_handler.py

```diff
@@ -785,6 +785,10 @@ def store_testcase(crash, fuzzed_keys, minimized_keys, regression, fixed,
   # Set metadata fields (e.g. build url, build key, platform string, etc).
   set_initial_testcase_metadata(testcase)
 
+  # Set crash metadata.
+  if hasattr(crash, 'crash_categories') and crash.crash_categories:
```

Comment:
Do we really need this hasattr check? I think it should always be around now right?

Reply:
Ah, this is here because corpus_pruning_task._process_corpus_crashes() calls store_testcase() with crash=CorpusCrash(), and CorpusCrash() does not have a crash_categories field. Earlier iterations of this PR *did* add crash_categories as a field to CorpusCrash(), but the added field required changes to untrusted_runner.proto and Testcase(). I removed the field when Oliver suggested in earlier comments against changing the Testcase() definition, resulting in this current PR and if-statement.

Reply:
Can you leave a TODO(#THIS COMMENTTHREAD) to get rid of this then please?

Reply:
Thanks - I've just added a new comment with a link to this thread.